src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -231,31 +271,28 @@ void SensorFusion::updateWorld(const SSLProto::SSL_DetectionFrame &ssl_detection
         ball_in_dribbler_timeout                = 0;
     }
 
-    if (friendly_robot_id_with_ball_in_dribbler.has_value())
+    if (shouldTrustRobotStatus(ssl_detection_frame, ball_detections))
     {
         std::optional<Robot> robot_with_ball_in_dribbler =
             friendly_team.getRobotById(friendly_robot_id_with_ball_in_dribbler.value());
 
-        if (robot_with_ball_in_dribbler.has_value())
+        std::vector<BallDetection> dribbler_in_ball_detection = {BallDetection{
+            .position =
+                robot_with_ball_in_dribbler->position() +
+                Vector::createFromAngle(robot_with_ball_in_dribbler->orientation())
+                    // MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT of the ball should be
+                    // inside the robot
```

Comment:
what is this comment referring to? is this supposed to be here?

Reply:
oh this was in the old code I see. you can delete this comment, I don't think it's relevant anymore